rfc/rfc-65/rfc-65.md

```diff
@@ -0,0 +1,248 @@
+    }
+  }
+}
+``` 
+
+### Executing partition lifecycle management
+
+Once we already have a proper `PartitionLifecycleManagementStrategy` implementation, it's easy to execute the partition
+lifecycle management.
+
+```java
+public class SparkPartitionLifecycleManagementActionExecutor<T> extends BaseSparkCommitActionExecutor<T> {
+  @Override
+  public HoodieWriteMetadata<HoodieData<WriteStatus>> execute() {
+    // Construct PartitionLifecycleManagementStrategy
+    PartitionLifecycleManagementStrategy strategy = (PartitionLifecycleManagementStrategy) ReflectionUtils.loadClass(
+        PartitionLifecycleManagementStrategy.checkAndGetPartitionLifecycleManagementStrategy(config),
+        new Class<?>[] {HoodieTable.class, HoodieEngineContext.class, HoodieWriteConfig.class}, table, context, config);
+
+    // Get expired partition paths
+    List<String> expiredPartitions = strategy.getExpiredPartitionPaths();
+
+    // Delete them reusing SparkDeletePartitionCommitActionExecutor
+    return new SparkDeletePartitionCommitActionExecutor<>(context, config, table, instantTime,
+        expiredPartitions).execute();
+  }
+}
+```
+
+We will add a new method `managePartitionLifecycle` in `HoodieTable` and `HoodieSparkCopyOnWriteTable` can implement it
+like this:
+
+```java
+@Override
+public HoodieWriteMetadata<HoodieData<WriteStatus>>managePartitionLifecycle(HoodieEngineContext context,String instantTime){
+    return new SparkPartitionLifecycleManagementActionExecutor<>(context,config,this,instantTime).execute();
+    }
+```
+
+We can call `hoodieTable.managePartitionLifecycle` in independent flink/spark job, in async/sync inline table services
+like clustering/compaction/clean etc.
+
+### User interface for Partition lifecycle Management
+
+We can do partition lifecycle management inline with streaming ingestion job or do it with a independent batch job, for
+both spark and flink engine.
+
+#### Run inline with Streaming Ingestion
+
+Since we can run clustering inline with streaming ingestion job through the following config:
+
+```properties
+hoodie.clustering.async.enabled=true
+hoodie.clustering.async.max.commits=5
+```
+
+We can do similar thing for partition lifecycle management. The config for async partition lifecycle management are:
+
+| Config key                                              | Remarks                                                                                                                        | Default |
+|---------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------|---------|
+| hoodie.partition.lifecycle.management.async.enabled     | Enable running of lifecycle management service, asynchronously as writes happen on the table.                                  | False   |
```

Comment:
SHould we handle the cleaning duty just to the cleaner service? Like we do to the `delete partition` cmds, you just generate a delete partiton commit metadata on the timeline and the cleaner would take care of it.

Reply:
What I really care about is the conflict resolution for the operation, does OCC applies here? Do we have some support for the basic concurrent modification.

Reply:
Yes, we will use `delete partiton` command to delete the expired partitions, this is mentioned in `Background` section. And conflict resolution for the operation is the same with what we do in `delete partiton` command.